File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    score = [0, 3, 2, 1, 0, 1, 2, 3]
    type = {'R': 0, 'T': 0, 'C': 0, 'F': 0, 'J': 0, 'M': 0, 'A': 0, 'N': 0}
    
    # 점수 계산
    for i in range(len(survey)):
        c = survey[i][0] if choices[i] < 4 else survey[i][1]
        type[c] += score[choices[i]]
        
    # 유형 판별
    for t in ["RT", "CF", "JM", "AN"]:
        answer += t[0] if type[t[0]] >= type[t[1]] else t[1]
        
    return answer

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        int[] score = { 0, 3, 2, 1, 0, 1, 2, 3 };
        String type = "RTCFJMAN";
        Map<String, Integer> typeCnt = new HashMap<>();
        for (int i = 0; i < type.length(); i++)
            typeCnt.put(Character.toString(type.charAt(i)), 0);
        
        // 점수 구하기
        for (int i = 0; i < survey.length; i++) {
            String c = Character.toString(choices[i] < 4 ? survey[i].charAt(0) : survey[i].charAt(1));
            typeCnt.put(c, typeCnt.get(c) + score[choices[i]]);
        }
        
        // 유형 판별
        for (int i = 0; i < 8; i += 2) {
            String f = Character.toString(type.charAt(i));
            String s = Character.toString(type.charAt(i + 1));
            answer += typeCnt.get(f) >= typeCnt.get(s) ? f : s;
        }
        
        return answer;
    }
}